pithos/pithos.py

```diff
@@ -90,21 +106,96 @@ def do_set_property(self, pspec, value):
     def do_get_property(self, pspec):
         return getattr(self, pspec.name)
     def do_render(self, ctx, widget, background_area, cell_area, flags):
-        if self.pixbuf:
+        if self.pixbuf is not None:
             Gdk.cairo_set_source_pixbuf(ctx, self.pixbuf, cell_area.x, cell_area.y)
             ctx.paint()
-        if self.icon:
+        else:
+            Gdk.cairo_set_source_pixbuf(ctx, self.background, cell_area.x, cell_area.y)
+            ctx.paint()
+            x = cell_area.x + (ALBUM_ART_SIZE - self.generic_audio_icon.get_width()) // 2
+            y = cell_area.y + (ALBUM_ART_SIZE - self.generic_audio_icon.get_height()) // 2
+            Gdk.cairo_set_source_pixbuf(ctx, self.generic_audio_icon, x, y)
+            ctx.paint()
+
+        if self.icon is not None:
             x = cell_area.x + (cell_area.width - self.rate_bg.get_width()) # right
             y = cell_area.y + (cell_area.height - self.rate_bg.get_height()) # bottom
             Gdk.cairo_set_source_pixbuf(ctx, self.rate_bg, x, y)
             ctx.paint()
 
-            pixbuf = Gtk.IconTheme.get_default().load_icon(self.icon, Gtk.IconSize.MENU, 0)
-            x = cell_area.x + (cell_area.width - pixbuf.get_width()) - 5 # right
-            y = cell_area.y + (cell_area.height - pixbuf.get_height()) - 5 # bottom
-            Gdk.cairo_set_source_pixbuf(ctx, pixbuf, x, y)
+            if self.icon == 'love':
+                rating_icon = self.love_icon
+            elif self.icon == 'tired':
+                rating_icon = self.tired_icon
+            elif self.icon == 'ban':
+                rating_icon = self.ban_icon
+
+            x = x + (rating_icon.get_width() // 2)
+            y = y + (rating_icon.get_height() // 2)
+
+            Gdk.cairo_set_source_pixbuf(ctx, rating_icon, x, y)
             ctx.paint()
 
+    def update_icons(self, style_context):
+        #Dynamically change the color of backgrounds and icons
```

Comment:
__[flake8]__
* [E265] block comment should start with '# '

<!-- Comment by [SideCI](https://sideci.com) -->